File: apiwdbe/WdbeQSilList.cpp
/**
	* \file WdbeQSilList.cpp
	* API code for table TblWdbeQSilList (implementation)
	* \author Alexander Wirthmueller
	* \date created: 11 Jul 2020
	* \date modified: 11 Jul 2020
	*/

#include "WdbeQSilList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQSilList
 ******************************************************************************/

WdbeQSilList::WdbeQSilList(
			const uint jnum
			, const string sref
			, const string Title
			, const string Fullsref
			, const string srefIxVBasetype
			, const string titIxVBasetype
			, const string srefRefIxVTbl
			, const string titRefIxVTbl
			, const string stubRefUref
			, const string stubRefWdbeMSystem
			, const string stubRefWdbeMModule
			, const string srefKPackage
			, const string titSrefKPackage
			, const string yesnoEasy
			, const string srefKToolch
			, const string titSrefKToolch
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->Title = Title;
	this->Fullsref = Fullsref;
	this->srefIxVBasetype = srefIxVBasetype;
	this->titIxVBasetype = titIxVBasetype;
	this->srefRefIxVTbl = srefRefIxVTbl;
	this->titRefIxVTbl = titRefIxVTbl;
	this->stubRefUref = stubRefUref;
	this->stubRefWdbeMSystem = stubRefWdbeMSystem;
	this->stubRefWdbeMModule = stubRefWdbeMModule;
	this->srefKPackage = srefKPackage;
	this->titSrefKPackage = titSrefKPackage;
	this->yesnoEasy = yesnoEasy;
	this->srefKToolch = srefKToolch;
	this->titSrefKToolch = titSrefKToolch;
};

bool WdbeQSilList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQSilList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "Title", "tit", Title);
		extractStringUclc(docctx, basexpath, "Fullsref", "fsr", Fullsref);
		extractStringUclc(docctx, basexpath, "srefIxVBasetype", "typ", srefIxVBasetype);
		extractStringUclc(docctx, basexpath, "titIxVBasetype", "typ2", titIxVBasetype);
		extractStringUclc(docctx, basexpath, "srefRefIxVTbl", "ret", srefRefIxVTbl);
		extractStringUclc(docctx, basexpath, "titRefIxVTbl", "ret2", titRefIxVTbl);
		extractStringUclc(docctx, basexpath, "stubRefUref", "reu", stubRefUref);
		extractStringUclc(docctx, basexpath, "stubRefWdbeMSystem", "sys", stubRefWdbeMSystem);
		extractStringUclc(docctx, basexpath, "stubRefWdbeMModule", "mdl", stubRefWdbeMModule);
		extractStringUclc(docctx, basexpath, "srefKPackage", "pkg", srefKPackage);
		extractStringUclc(docctx, basexpath, "titSrefKPackage", "pkg2", titSrefKPackage);
		extractStringUclc(docctx, basexpath, "yesnoEasy", "esy", yesnoEasy);
		extractStringUclc(docctx, basexpath, "srefKToolch", "tch", srefKToolch);
		extractStringUclc(docctx, basexpath, "titSrefKToolch", "tch2", titSrefKToolch);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQSilList
 ******************************************************************************/

ListWdbeQSilList::ListWdbeQSilList() {
};

ListWdbeQSilList::ListWdbeQSilList(
			const ListWdbeQSilList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQSilList(*(src.nodes[i]));
};

ListWdbeQSilList& ListWdbeQSilList::operator=(
			const ListWdbeQSilList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQSilList(*(src.nodes[i]));

	return *this;
};

ListWdbeQSilList::~ListWdbeQSilList() {
	clear();
};

void ListWdbeQSilList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQSilList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQSilList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQSilList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQSilList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQSilList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQSilList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwdbe/WdbeQSilList.java
/**
	* \file WdbeQSilList.h
	* Java API code for record of table TblWdbeQSilList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQSilList {
	
	public WdbeQSilList(
				int jnum
				, String sref
				, String Title
				, String Fullsref
				, String srefIxVBasetype
				, String titIxVBasetype
				, String srefRefIxVTbl
				, String titRefIxVTbl
				, String stubRefUref
				, String stubRefWdbeMModule
				, String srefKPackage
				, String titSrefKPackage
				, String yesnoEasy
				, String srefKToolch
				, String titSrefKToolch
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.Title = Title;
		this.Fullsref = Fullsref;
		this.srefIxVBasetype = srefIxVBasetype;
		this.titIxVBasetype = titIxVBasetype;
		this.srefRefIxVTbl = srefRefIxVTbl;
		this.titRefIxVTbl = titRefIxVTbl;
		this.stubRefUref = stubRefUref;
		this.stubRefWdbeMModule = stubRefWdbeMModule;
		this.srefKPackage = srefKPackage;
		this.titSrefKPackage = titSrefKPackage;
		this.yesnoEasy = yesnoEasy;
		this.srefKToolch = srefKToolch;
		this.titSrefKToolch = titSrefKToolch;
	};

	public int jnum;
	public String sref;
	public String Title;
	public String Fullsref;
	public String srefIxVBasetype;
	public String titIxVBasetype;
	public String srefRefIxVTbl;
	public String titRefIxVTbl;
	public String stubRefUref;
	public String stubRefWdbeMModule;
	public String srefKPackage;
	public String titSrefKPackage;
	public String yesnoEasy;
	public String srefKToolch;
	public String titSrefKToolch;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQSilList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			Title = Xmlio.extractStringUclc(doc, basexpath, "Title", "tit", null, 0);
			Fullsref = Xmlio.extractStringUclc(doc, basexpath, "Fullsref", "fsr", null, 0);
			srefIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "srefIxVBasetype", "typ", null, 0);
			titIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "titIxVBasetype", "typ2", null, 0);
			srefRefIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "srefRefIxVTbl", "ret", null, 0);
			titRefIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "titRefIxVTbl", "ret2", null, 0);
			stubRefUref = Xmlio.extractStringUclc(doc, basexpath, "stubRefUref", "reu", null, 0);
			stubRefWdbeMModule = Xmlio.extractStringUclc(doc, basexpath, "stubRefWdbeMModule", "mdl", null, 0);
			srefKPackage = Xmlio.extractStringUclc(doc, basexpath, "srefKPackage", "pkg", null, 0);
			titSrefKPackage = Xmlio.extractStringUclc(doc, basexpath, "titSrefKPackage", "pkg2", null, 0);
			yesnoEasy = Xmlio.extractStringUclc(doc, basexpath, "yesnoEasy", "esy", null, 0);
			srefKToolch = Xmlio.extractStringUclc(doc, basexpath, "srefKToolch", "tch", null, 0);
			titSrefKToolch = Xmlio.extractStringUclc(doc, basexpath, "titSrefKToolch", "tch2", null, 0);
			
			return true;
		};

		return false;
	};
};
